x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/pages/coverage_overview/helpers.ts

```diff
@@ -22,3 +32,39 @@ export const getCardBackgroundColor = (value: number) => {
     }
   }
 };
+
+export const formatRuleFilterOptions = <
+  T extends CoverageOverviewRuleSource | CoverageOverviewRuleActivity
+>(
+  options: Array<{ checked?: string; label: T }>
+): T[] => {
+  return options.filter((option) => option.checked === 'on').map((option) => option.label);
+};
+
+export const getInitialRuleActivityFilterOptions = (
+  filter: CoverageOverviewFilter
+): EuiSelectableOption[] =>
+  ruleActivityFilterDefaultOptions.map((option) => {
+    if (filter.activity?.includes(option.label)) {
+      return { ...option, checked: 'on' };
+    }
+    return option;
+  });
+
+export const getInitialRuleSourceFilterOptions = (
+  filter: CoverageOverviewFilter
+): EuiSelectableOption[] =>
+  ruleSourceFilterDefaultOptions.map((option) => {
+    if (filter.source?.includes(option.label)) {
+      return { ...option, checked: 'on' };
+    }
+    return option;
+  });
+
+export const getInitialFilterOptions = (
```

Comment:
nit: this function doesn't know anything about the filter so using `Filter` in name doesn't add readability. The function only marks selected EuiSelectable options with `checked: 'on'`. Something like `markSelected()` would reflect the intention better.